connectors/sources/github.py

```diff
@@ -33,57 +120,115 @@ def __init__(self, configuration):
         self.configuration = configuration
         self.repos = self.configuration["repositories"]
         self.github_token = self.configuration["github_token"]
-        self.headers = {"Authorization": f"Bearer {self.github_token}"}
+        self.headers = {
+            "Authorization": f"Bearer {self.github_token}",
+            "Accept": "application/vnd.github.raw",
+        }
 
-    async def get_item(self, resource):
+    async def _get_retry_after(self):
+        response = await self._get_client.getitem("/rate_limit")
+        reset = response["resources"]["core"]["reset"]
+        # Adding a 5 second delay to account for server delays
+        return (reset - time.time()) + 5
+
+    async def _put_to_sleep(self):
+        retry_after = await self._get_retry_after()
+        logger.debug(f"Connector will attempt to retry after {retry_after} seconds.")
+        await self._sleeps.sleep(retry_after)
+        raise Exception("Rate limit exceeded.")
```

Comment:
Should we always raise, when we call that method?

Reply:
yes, to trigger `retryable`